homeassistant/components/go2rtc/__init__.py

```diff
@@ -99,19 +114,47 @@ async def on_stop(event: Event) -> None:
 
         url = "http://localhost:1984/"
 
+    hass.data[_DATA_GO2RTC] = url
+    discovery_flow.async_create_flow(
+        hass, DOMAIN, context={"source": SOURCE_SYSTEM}, data={}
+    )
+    return True
+
+
+async def _remove_go2rtc_entries(hass: HomeAssistant) -> None:
+    """Remove go2rtc config entries, if any."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        await hass.config_entries.async_remove(entry.entry_id)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up go2rtc from a config entry."""
+    url = hass.data[_DATA_GO2RTC]
+
     # Validate the server URL
     try:
         client = Go2RtcRestClient(async_get_clientsession(hass), url)
         await client.streams.list()
-    except Exception:  # noqa: BLE001
-        _LOGGER.warning("Could not connect to go2rtc instance on %s", url)
+    except Go2RtcClientError as err:
+        if isinstance(err.__cause__, _RETRYABLE_ERRORS):
+            _LOGGER.info("Could not connect to go2rtc instance on %s, will retry", url)
```

Comment:
Pass the message to the ConfigEntryNotReady exception as an argument instead of logging it. It will be logged by default.